onegov/swissvotes/models/vote.py

```diff
@@ -559,24 +560,67 @@ def results_cantons(self):
 
     def get_recommendation(self, name):
         """ Get the recommendations by name. """
+        # Todo: remove inconsistencies in table name
+        name_ = name
+        return self.codes('recommendation').get(
+            self.recommendations.get(name_)
+        )
 
-        recommendations = self.recommendations or {}
-        return self.codes('recommendation').get(recommendations.get(name))
+    def get_recommendation_of_existing_parties(self):
+        """ Get only the existing parties as when this vote was conducted """
+        if not self.recommendations:
+            return {}
+        return {k: v for k, v in self.recommendations.items() if v != 9999}
 
     def group_recommendations(self, recommendations):
         """ Group the given recommendations by slogan. """
 
+        codes = self.codes('recommendation')
+        recommendation_codes = list(codes.keys())
+
+        def by_recommendation(reco):
+            return recommendation_codes.index(reco)
+
         result = {}
         for actor, recommendation in recommendations:
-            if recommendation is not None and recommendation != 9999:
+            if recommendation != 9999:
                 result.setdefault(recommendation, []).append(actor)
 
-        codes = self.codes('recommendation')
         return OrderedDict([
             (codes[key], result[key])
-            for key in sorted(result.keys())
+            for key in sorted(result.keys(), key=by_recommendation)
         ])
 
+    def get_actors_share(self, actor):
+        assert isinstance(actor, str), 'Actor must be a string'
+        actor_ = actor
```

Comment:
I would introduce a method for this. For example `normalize_actor_shortcut(shortcut)`. In it a longer comment could mention the less-than-desirable state of the database as well as the todo comment.

Minor nitpick about todos, btw: You should write them like this: `# TODO:` or `# FIXME:` or `# XXX:`, as those kinds of strings are picked up by various editors and highlighted (but only if they are all uppercase).